### DISCUSSION TITLE
Type for Prisma query results

### DISCUSSION THREAD

**Discussion by davesteinberg** [prisma/prisma#7377] [Q&A] [answered]
This may be more of a TypeScript question than a Prisma question, but I'm new to both and Prisma's typings are definitely next-level.

I'm looking for a DRY approach to define a type for the result of a Prisma query. The query will run on the server (naturally), but I want to use the type to write typed client code that deals with the data its getting from the server.

Here's a relevant snippet of my schema:

```
model Story {
  id             String      @id @default(cuid())
  createdAt      DateTime    @default(now())
  updatedAt      DateTime    @updatedAt
  content        String
  postal         String
  postalCode     PostalCode? @relation(fields: [postal], references: [code])
  displayName    String?
  contactName    String?
  email          String?
  approved       Boolean     @default(false)
}
```

My server-side query is like this:

```
export const select = {
  id: true,
  createdAt: true,
  updatedAt: true,
  content: true,
  postal: true,
  postalCode: true,
  displayName: true,
  approved: true
}

...
return await prisma.story.findMany({
  where: { approved: true },
  orderBy: { updatedAt: 'desc' },
  select,
})
```

So, you can see I'm returning a subset of the `Story` fields and also including a `PostalCode` entity via a relation.  I want to define a type, which for now we'll call a `StoryResult`, that I can use in client code like this:

```
export function storyCite({ displayName, postal, postalCode }: StoryResult): string {
  ...
}
``` 

I can't just use the plain `Story` type that's generated by Prisma because it doesn't include `postalCode`. I see there's a `StoryGetPayload` in the generated types, but I can't figure out how to use it.

Any pointers would be much appreciated.

**Comment by davesteinberg** [comment, marked as answer]
Update: With a bit more search, I found [this answer](https://github.com/prisma/prisma/discussions/4372#discussioncomment-139252). So, now I've got this:

```
import { Prisma } from '@prisma/client'

export const select = {
  id: true,
  createdAt: true,
  updatedAt: true,
  content: true,
  postal: true,
  postalCode: true,
  displayName: true,
  approved: true
}

export type Story = Prisma.StoryGetPayload<{ select: typeof select }>
```

It seems to be doing the trick. Thanks @nikolasburk for the super helpful answer!

**Reply by yanmarinich** [reply]
export const select = {
  id: true,
  createdAt: true,
  updatedAt: true,
  content: true,
  postal: true,
  postalCode: true,
  displayName: true,
  approved: true
} satisfies Prisma.StorySelect

this one will narrow type without any extra variables and you can use dto from the start

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
